### PR TITLE
Added auth check for fund

### PR DESCRIPTION
**Pull request by smartdevcom**
Added `auth` check for supply and borrow. In all chains except NEON, user should be authorized to supply or borrow and this should be checked before any actions.In NEON, every user is allowed to do any actions by default.

Tested:
`0x8982aa50bb919E42e9204f12e5b59D053Eb2A602`: doesn't show auth message for borrow
`0x2924973E3366690eA7aE3FCdcb2b4e136Cf7f8Cc`: shows auth message for borrow

For supply, it doesn't show msg because all accounts are allowed to supply in chapel

### CODE REVIEW

File: packages/sdk/src/modules/Pools.ts [hunk 1]
```diff
@@ -229,5 +232,31 @@ export function withPools<TBase extends IonicBaseConstructor>(Base: TBase) {
 
       return [...filteredPools, ...whitelistedPools].filter((p) => !!p) as IonicPoolData[];
     }
+
+    async isAuth(pool: string, market: string, role: Roles, user: string) {
+      const authRegistry = this.createAuthoritiesRegistry();
+      const poolAuthAddress = await authRegistry.callStatic.poolsAuthorities(pool);
+
+      if (poolAuthAddress === constants.AddressZero && this.chainId !== SupportedChains.neon) {
```
Comment (carlomazzaferro) [inline packages/sdk/src/modules/Pools.ts:240]: This is wrong. If the chainId  == neon.chainId, then `isAuth` must return True.

Comment (smartdevcom) [reply packages/sdk/src/modules/Pools.ts]: fixed

**Comment by carlomazzaferro** [comment]
Ok, let's fix CI now

**Comment by smartdevcom** [comment]
> Ok, let's fix CI now

https://github.com/ionicprotocol/monorepo/actions/runs/5936041680/job/16095648676?pr=63
Isn't this related to npm token?

**Comment by carlomazzaferro** [comment]
> > Ok, let's fix CI now
> 
> https://github.com/ionicprotocol/monorepo/actions/runs/5936041680/job/16095648676?pr=63
> Isn't this related to npm token?

No, it isn't

**Pull request merged**